src/dmd/expressionsem.d

```diff
@@ -3518,16 +3521,18 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             {
                 exp.f = exp.f.toAliasFunc();
                 TypeFunction tf = cast(TypeFunction)exp.f.type;
-                if (!tf.callMatch(null, exp.arguments))
+                size_t failIndex;
+                if (!tf.callMatch(null, exp.arguments, 0, &failIndex))
                 {
                     OutBuffer buf;
                     buf.writeByte('(');
                     argExpTypesToCBuffer(&buf, exp.arguments);
                     buf.writeByte(')');
 
                     //printf("tf = %s, args = %s\n", tf.deco, (*arguments)[0].type.deco);
-                    .error(exp.loc, "`%s%s` is not callable using argument types `%s`", exp.e1.toChars(), parametersTypeToChars(tf.parameters, tf.varargs), buf.peekString());
-
+                    .error(exp.loc, "`%s%s` is not callable using argument types `%s`",
+                        exp.e1.toChars(), parametersTypeToChars(tf.parameters, tf.varargs), buf.peekString());
+                    showArgMismatch(exp.loc, exp.arguments, tf, failIndex);
```

Comment:
It'd be nice to have test coverage for the new code.

Reply:
BTW, what does TOKvar mean? I'm trying to hit this case.

Reply:
`TOKvar` distinguishes a `VarExp` from other Expression types.

Reply:
I don't know how to test for the case when TOKvar is a function. Function pointers are handled differently.